packages/store/src/module.ts

```diff
@@ -1,108 +1,39 @@
 import { LifecycleStateManager } from './internal/lifecycle-state-manager';
-import { InternalDispatcher, InternalDispatchedActionResults } from './internal/dispatcher';
+import { InternalDispatchedActionResults, InternalDispatcher } from './internal/dispatcher';
 import { InternalStateOperations } from './internal/state-operations';
 import { Store } from './store';
 import { SelectFactory } from './decorators/select';
 import { StateStream } from './internal/state-stream';
 import { PluginManager } from './plugin-manager';
-import { InitState, UpdateState } from './actions/actions';
-import { StateClass } from './internal/internals';
+import {
+  appBootstrapListenerFactory,
+  ngxsConfigFactory,
+  StateClass
+} from './internal/internals';
+import { NgxsRootModule } from './modules/ngxs-root.module';
+import { NgxsFeatureModule } from './modules/ngxs-feature.module';
 import { DispatchOutsideZoneNgxsExecutionStrategy } from './execution/dispatch-outside-zone-ngxs-execution-strategy';
 import { InternalNgxsExecutionStrategy } from './execution/internal-ngxs-execution-strategy';
 
-/**
- * Root module
- * @ignore
- */
-@NgModule()
-export class NgxsRootModule {
-  constructor(
-    factory: StateFactory,
-    internalStateOperations: InternalStateOperations,
-    store: Store,
-    select: SelectFactory,
-    @Optional()
-    @Inject(ROOT_STATE_TOKEN)
-    states: StateClass[] = [],
-    lifecycleStateManager: LifecycleStateManager
-  ) {
-    // add stores to the state graph and return their defaults
-    const results = factory.addAndReturnDefaults(states);
-
-    internalStateOperations.setStateToTheCurrentWithNew(results);
-
-    // connect our actions stream
-    factory.connectActionHandlers();
-
-    // dispatch the init action and invoke init and bootstrap functions after
-    lifecycleStateManager.ngxsBootstrap(new InitState(), results);
-  }
-}
-
-/**
- * Feature module
- * @ignore
- */
-@NgModule({})
-export class NgxsFeatureModule {
-  constructor(
-    store: Store,
-    internalStateOperations: InternalStateOperations,
-    factory: StateFactory,
-    @Optional()
-    @Inject(FEATURE_STATE_TOKEN)
-    states: any[][],
-    lifecycleStateManager: LifecycleStateManager
-  ) {
-    // Since FEATURE_STATE_TOKEN is a multi token, we need to
-    // flatten it [[Feature1State, Feature2State], [Feature3State]]
-    const flattenedStates = ([] as any[]).concat(...states);
-
-    // add stores to the state graph and return their defaults
-    const results = factory.addAndReturnDefaults(flattenedStates);
-
-    if (results.states.length) {
-      internalStateOperations.setStateToTheCurrentWithNew(results);
-      // dispatch the update action and invoke init and bootstrap functions after
-      lifecycleStateManager.ngxsBootstrap(new UpdateState(), results);
-    }
-  }
-}
-
-export type ModuleOptions = Partial<NgxsConfig>;
-
-export function ngxsConfigFactory(options: ModuleOptions): NgxsConfig {
```

Comment:
Both these factories (`ngxsConfigFactory` and `appBootstrapListenerFactory`) are better kept in the same file as the `NgxsModule` because they aren't used from anywhere else. Also, moving these away could cause problems with AOT compilation (...or maybe that has been fixed in the Angular compiler now).

I think that the `ModuleOptions` and the `ROOT_OPTIONS` should also remain in this file because they are concepts that are local to the bootstrapping methods in this file (they are only exported to satisfy the Angular compiler).

Reply:
Why do we have to move the ModuleOptions file? It should be public (expose from symbol.ts), as we describe it in the documentation for use it.